chromium_src/chrome/browser/browser_about_handler.cc

```diff
@@ -1,34 +1,36 @@
-/* This Source Code Form is subject to the terms of the Mozilla Public
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
  * License, v. 2.0. If a copy of the MPL was not distributed with this file,
  * You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 #define FixupBrowserAboutURL FixupBrowserAboutURL_ChromiumImpl
-#define WillHandleBrowserAboutURL WillHandleBrowserAboutURL_ChromiumImpl
-#include "../../../../chrome/browser/browser_about_handler.cc"
+#include "../../../../chrome/browser/browser_about_handler.cc"  // NOLINT
 #undef FixupBrowserAboutURL
-#undef WillHandleBrowserAboutURL
 
 #include "brave/common/url_constants.h"
+#include "brave/common/webui_url_constants.h"
 
-bool FixupBrowserAboutURLBraveImpl(GURL* url,
-                                   content::BrowserContext* browser_context) {
-  if (url->SchemeIs(kBraveUIScheme)) {
+bool FixupBrowserAboutURL(GURL* url,
+                          content::BrowserContext* browser_context) {
+  bool result = FixupBrowserAboutURL_ChromiumImpl(url, browser_context);
+
+  // no special win10 welcome page
+  if (url->host() == chrome::kChromeUIWelcomeWin10Host)
+    *url = GURL(chrome::kChromeUIWelcomeURL);
+
+  // redirect sync-internals
+  if (url->host() == chrome::kChromeUISyncInternalsHost) {
     GURL::Replacements replacements;
-    replacements.SetSchemeStr(content::kChromeUIScheme);
+    replacements.SetHostStr(chrome::kChromeUISyncHost);
     *url = url->ReplaceComponents(replacements);
   }
-  return true;
-}
-
 
-bool FixupBrowserAboutURL(GURL* url,
-                          content::BrowserContext* browser_context) {
-  FixupBrowserAboutURLBraveImpl(url, browser_context);
-  return FixupBrowserAboutURL_ChromiumImpl(url, browser_context);
-}
+  if (url->SchemeIs(content::kChromeUIScheme) &&
+      url->host() != chrome::kChromeUINewTabHost) {
```

Comment:
Just curiosity, any particular reason about excluding newtab host?

Reply:
yes, it breaks the extension newtab override